src/List/List.js

```diff
@@ -45,13 +49,16 @@ export default class List extends Component {
       className: classNameProp,
       component,
       padding,
+      children,
       ...other,
     } = this.props;
     const classes = this.context.styleManager.render(styleSheet);
+    const firstChild = Children.toArray(children)[0];
     const className = classNames(classes.root, {
       [classes.padding]: padding,
+      [classes.subheader]: isValidElement(firstChild) && firstChild.type === Subheader,
```

Comment:
Yes, I have the [`<ListItem />`](https://github.com/callemall/material-ui/blob/next/src/List/ListItem.js#L100-L102)